pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -50,9 +52,10 @@ let
 
     CC_host = "cc";
     CXX_host = "c++";
-    depsBuildBuild = [ buildPackages.stdenv.cc openssl libuv zlib ];
+    depsBuildBuild = [ buildPackages.stdenv.cc openssl libuv zlib icu ];
 
     buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ApplicationServices ]
+      ++ lib.optional isCross runtimeShellPackage # for patchShebangs
```

Comment:
Actually one more comment too: is this just meant to be `bash` and if it is needed at runtime, shouldn't it just be unconditionally added instead of only added for cross?

Reply:
Yes, bash/sh. I think it should be possible to remove bash/sh dependency since all scripts seem to be simple wrappers that invoke node on the corresponding JS script. I’ve tried that, but IIRC some other packages use node-gyp shell script wrapper and I didn’t investigate failures.

<details>
<summary><code>rg --sort path /nix/store</code></summary>

```
include/node/config.gypi
2:{ 'target_defaults': {'cflags': [], 'default_configuration': 'Release', 'defines': ['NODE_OPENSSL_CONF_NAME=nodejs_conf', 'ICU_NO_USER_DATA_OVERRIDE'], 'include_dirs': ['/nix/store/v1ivffsk8kcnzmkanf1hdw8jdhf8cnni-zlib-1.2.13-dev/include', '/nix/store/16f23j7k72rdjji6g0106h2ss1ghx3ys-libuv-1.45.0-dev/include', '/nix/store/xsc0lsv634lv5zk9dfdfaysv3x79y52r-openssl-3.0.9-dev/include', '/nix/store/viagaap65dxbijv0pahim3dkzkakhlpa-icu4c-73.1-dev/include'], 'libraries': ['-L/nix/store/lxgxpd3yl9y8mwkqs905lms4vzwca3hy-zlib-1.2.13/lib', '-lz', '-L/nix/store/0m9z7g2f5y42w19jxykw9rhmaqg25arj-libuv-1.45.0/lib', '-luv', '-L/nix/store/xim52l8jwiasmmalsnwcy51f3l77cggm-openssl-3.0.9/lib', '-lcrypto', '-lssl', '-L/nix/store/h79hac9j683jcm3q0fky6kbj4flnivbh-icu4c-73.1/lib', '-licui18n', '-licuuc', '-licudata']},
337:                 'node_prefix': '/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0',

lib/node_modules/corepack/dist/corepack.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/npm.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/npx.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/pnpm.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/pnpx.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/yarn.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/yarnpkg.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/shims/corepack
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/npm
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/npx
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/pnpm
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/pnpx
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/yarn
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/yarnpkg
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/bin/node-gyp-bin/node-gyp
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/bin/npm
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/bash

lib/node_modules/npm/bin/npm-cli.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/bin/npx
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/bash

lib/node_modules/npm/bin/npx-cli.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/lib/utils/completion.sh
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/bash

lib/node_modules/npm/node_modules/@npmcli/arborist/bin/index.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/@npmcli/run-script/lib/node-gyp-bin/node-gyp
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/node_modules/color-support/bin.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/cssesc/bin/cssesc
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/mkdirp/bin/cmd.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/node-gyp/gyp/gyp
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/node_modules/node-gyp/gyp/tools/emacs/run-unit-tests.sh
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/node_modules/node-gyp/node_modules/nopt/bin/nopt.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/node-gyp/node_modules/which/bin/node-which
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/nopt/bin/nopt.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/pacote/lib/bin.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/qrcode-terminal/bin/qrcode-terminal.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/rimraf/bin.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/semver/bin/semver.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/sigstore/bin/sigstore.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/which/bin/which.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node
```
</details>

Also `HOST_PATH` in stdenv.mkDerivation is empty when cross-compiling (without `buildInputs`) but contains implicit coreutils, bash and other stuff when `hostPlatform == buildPlatform`. Not sure if this is intended behavior 🤷

Reply:
A bash dependency is fine, I'm just meaning why would it only be relevant for cross?

If it just pulling in the implicit bash from `nativeBuildInputs` for non-cross since `strictDeps == false`, we should be uncoditionally adding it rather than relying on implicit dependencies

Reply:
Agreed, should I also set `strictDeps = true`?

Reply:
Please make it unconditional so that we notice early if cross breaks in an update

Reply:
Done 8aaec56348450d639db2a700b178f5715b993976